subprojects/docs/src/snippets/java/toolchain-task/groovy/build.gradle

```diff
@@ -2,16 +2,16 @@ import javax.inject.Inject;
 
 class CustomTaskUsingToolchains extends DefaultTask {
 
-    @Input
-    final Property<JavaLauncher> launcher = project.objects.property(JavaLauncher.class);
+    @Nested
+    final Property<JavaLauncher> launcher = project.objects.property(JavaLauncher.class)
```

Comment:
Is there any reason we don't use abstract getters in the sample?

Reply:
Not that I'm aware of. Would that be our general recommendation? I personally find dealing with abstract methods/abstract class for the sake of decoration/injection quite confusing for a concrete implementation.

Reply:
That is at least my recommendation. I think it is much better to declare an abstract getter than accessing the project. You would end up with something like this:
```
@Nested
Property<JavaLauncher> getLauncher()
```

You can still access the property in Groovy as you did before.

Reply:
Fair enough, fixed